### PR TITLE
[WIP] Feature/aws lambda plugin

### PR DESCRIPTION
**Pull request by tomek-ifm**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/aws-lambda/src/components/useBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React, { SyntheticEvent, useEffect, useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Snackbar from '@material-ui/core/Snackbar';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+
+export default function SimpleBanner({
```
Comment (iain-b) [inline plugins/aws-lambda/src/components/useBanner.tsx:7]: Filename should match component name - SimpleBanner.tsx.

File: plugins/aws-lambda/src/components/AWSLambdaPageTable/AWSLambdaPageTable.tsx [hunk 2]
```diff
@@ -218,20 +229,48 @@ export const AWSLambdaPageTable = () => {
     );
   }, [tableProps.lambdaData, page, pageSize, lambdaAnnotationList]); // eslint-disable-line react-hooks/exhaustive-deps
 
+  const SettingsComponent = () => (
```
Comment (iain-b) [inline plugins/aws-lambda/src/components/AWSLambdaPageTable/AWSLambdaPageTable.tsx:232]: SettingsComponent deserves its own file.

File: plugins/aws-lambda/src/components/useProjectName.ts [hunk 3]
```diff
@@ -23,10 +23,18 @@ export const useProjectName = (name: EntityCompoundName) => {
...
-    );
+      [];
+    const projectName =
+      entity?.metadata.annotations?.['github.com/project-slug'] ?? '';
+
+    return { lambdaNames, projectName };
   });
-  return { value, loading, error };
+  return {
+    value: value?.projectName,
```
Comment (iain-b) [inline plugins/aws-lambda/src/components/useProjectName.ts:35]: I think the name of this hook is a bit too specific. Given we're likely to add more perhaps something like useServiceEntityAnnotations ? Also, we should move this file out of components and into a src/hooks directory.

**Review by iain-b — COMMENTED** [review]
Just minor comments on naming / folder structure.

**Review by iain-b — APPROVED** [review]

**Comment by iain-b** [comment]
Build is failing with:
```plugins/aws-lambda/dev/index.tsx:33:9 - error TS2345: Argument of type '{ apiOrigin: string; basePath: string; }' is not assignable to parameter of type '{ discoveryApi: DiscoveryApi; }'.
  Object literal may only specify known properties, and 'apiOrigin' does not exist in type '{ discoveryApi: DiscoveryApi; }'.

33         apiOrigin: 'http://localhost:7000',
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.```

**Comment by dtuite** [comment]
I haven't read the code but I will approve based on the conversations in standup.

**Review by dtuite — APPROVED** [review]

**Pull request closed without merging**